Sources/CorePayments/Networking/APIClient.swift

```diff
@@ -78,12 +79,15 @@ public class APIClient {
     
     // MARK: - Private Methods
     
-    private func constructRESTURL(path: String, queryParameters: [String: String]) throws -> URL {
+    private func constructRESTURL(path: String, queryParameters: [String: String]?) throws -> URL {
         let urlString = coreConfig.environment.baseURL.appendingPathComponent(path)
         var urlComponents = URLComponents(url: urlString, resolvingAgainstBaseURL: false)
         
-        queryParameters.forEach {
-            urlComponents?.queryItems?.append(URLQueryItem(name: $0.key, value: $0.value))
+        if let queryParameters {
+            urlComponents?.queryItems = []
+            queryParameters.forEach {
+                urlComponents?.queryItems?.append(URLQueryItem(name: $0.key, value: $0.value))
+            }
```

Comment:
```suggestion
        urlComponents?.queryItems = queryParameters?.map { URLQueryItem(name: $0.key, value: $0.value }
```
take or leave: we could use map here to streamline.

Reply:
Great idea - done in f43ce8218fa5f1e0682fe0a53c31f778526f2919